src/cdn_definitions/_impl/__init__.py

```diff
@@ -6,16 +6,46 @@
 import yaml
 from urllib3.util.retry import Retry
 
+# It is preferred to return the data as immutable when possible
+try:
+    from frozendict import frozendict
+except ImportError:
```

Comment:
Could you please mark the legacy code paths with `# pragma: no cover` so they're not dragging the coverage down?  (It's technically possible to merge the py2 and py3 coverage data instead, but I doubt it's worthwhile here.)